plugins/base/src/main/kotlin/translators/documentables/DescriptionSections.kt

```diff
@@ -209,25 +210,47 @@ internal fun PageContentBuilder.DocumentableContentBuilder.samplesSectionContent
 
 internal fun PageContentBuilder.DocumentableContentBuilder.inheritorsSectionContent(
     documentable: Documentable,
-    logger: DokkaLogger
+    logger: DokkaLogger,
 ) {
     val inheritors = if (documentable is WithScope) documentable.inheritors() else return
     if (inheritors.values.none()) return
 
-    // split content sections to this 2 different functions for 2 cases:
-    // * parent in the common code only and inheritor on the platform code
-    // * flag `mergeImplicitExpectActualDeclarations` used -> each instance will go through rendering
-    if (documentable.sourceSets.size == 1) singlePlatformInheritorsSectionContent(inheritors, logger)
-    else multiplatformInheritorsSectionContent(inheritors, logger)
+    // split content section for the case:
+    // parent is in the shared source set (without expect-actual) and inheritor is in the platform code
+    if (documentable.isDefinedInSharedSourceSetOnly(inheritors.keys.toSet()))
+        sharedSourceSetOnlyInheritorsSectionContent(inheritors, logger)
+    else
+        multiplatformInheritorsSectionContent(documentable, inheritors, logger)
 }
 
+/**
+ * Detect that documentable is located only in the shared code without expect-actuals
+ * Value of `analysisPlatform` will be [Platform.common] in cases if a source set shared between 2 different platforms.
+ * But if it shared between 2 same platforms (e.g. jvm("awt") and jvm("android"))
+ * then the source set will be still marked as jvm platform.
+ *
+ * So, we also try to check if any of inheritors source sets depends on current documentable source set.
+ * that will mean that the source set is shared.
+ */
+private fun Documentable.isDefinedInSharedSourceSetOnly(inheritorsSourceSets: Set<DokkaConfiguration.DokkaSourceSet>) =
+    sourceSets.size == 1 &&
+            (sourceSets.first().analysisPlatform == Platform.common
+                    || sourceSets.first().hasDependentSourceSetInInheritors(inheritorsSourceSets))
+
+private fun DokkaConfiguration.DokkaSourceSet.hasDependentSourceSetInInheritors(
+    inheritorsSourceSets: Set<DokkaConfiguration.DokkaSourceSet>,
+) =
+    inheritorsSourceSets.any { sourceSet -> sourceSet.dependentSourceSets.any { it == this.sourceSetID } }
+
 private fun PageContentBuilder.DocumentableContentBuilder.multiplatformInheritorsSectionContent(
+    documentable: Documentable,
     inheritors: Map<DokkaConfiguration.DokkaSourceSet, List<DRI>>,
-    logger: DokkaLogger
+    logger: DokkaLogger,
 ) {
-    val availablePlatforms = inheritors.keys.toSet()
+    // intersect is used for removing duplication in case of merged classlikes from different platforms
+    val availableSourceSets = inheritors.keys.toSet().intersect(documentable.sourceSets)
 
-    availablePlatforms.forEach { platform ->
+    availableSourceSets.forEach { platform ->
```

Comment:
`availableSourceSets.forEach` is not needed here (only in the table body) and in another function.

Reply:
My comment from the previous review is still actual.